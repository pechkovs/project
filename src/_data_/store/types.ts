export const bar = 'bar'
export const line = 'line'
export const pie = 'Pie'
