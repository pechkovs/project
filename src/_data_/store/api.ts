import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { getConfigValue } from '@ijl/cli'

type AdminStatRequestResult = {
    monthID: string
    amount: number
}[]

const BaseApiUrl = getConfigValue('sugarbun.baseApiUrl')

export const api = createApi({
    baseQuery: fetchBaseQuery({
        baseUrl: BaseApiUrl + '/admin/',
    }),
    reducerPath: 'admin',
    tagTypes: ['Orders', 'Sales', 'Revenue'],
    endpoints: (build) => ({
        getSales: build.query<AdminStatRequestResult, undefined>({
            query: () => 'sales/sales',
            providesTags: ['Sales'],
        }),
        getOrders: build.query<AdminStatRequestResult, undefined>({
            query: () => 'orders/orders',
            providesTags: ['Orders'],
        }),
        getRevenue: build.query<AdminStatRequestResult, undefined>({
            query: () => `revenue/revenue/`,
            providesTags: ['Revenue'],
        }),
    }),
})

export const { useGetSalesQuery, useGetOrdersQuery, useGetRevenueQuery } = api
