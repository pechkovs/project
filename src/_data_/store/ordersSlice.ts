import { createSelector, createSlice } from '@reduxjs/toolkit'

type ChartState = {
    mode: string
}

const initialState: ChartState = {
    mode: 'Pie',
}

const ordersChartSlice = createSlice({
    name: 'orders',
    initialState: initialState,
    reducers: {
        switchMode(state, action) {
            if (action.payload === 'Pie') {
                state.mode = 'Pie'
            }
            if (action.payload === 'Bar') {
                state.mode = 'Bar'
            }
            if (action.payload === 'Line') {
                state.mode = 'Line'
            }
        },
    },
})

export const ordersChartActions = ordersChartSlice.actions
export const ordersChartReducer = ordersChartSlice.reducer
export const ordersStateSelector = (state) => state.orders
export const ordersChartModeSelector = createSelector(
    ordersStateSelector,
    (ordersState) => ordersState.mode
)
export const isPieOrdersChart = createSelector(
    ordersChartModeSelector,
    (mode) => mode === 'Pie'
)
export const isBarOrdersChart = createSelector(
    ordersChartModeSelector,
    (mode) => mode === 'Bar'
)
export const isLineOrdersChart = createSelector(
    ordersChartModeSelector,
    (mode) => mode === 'Line'
)
