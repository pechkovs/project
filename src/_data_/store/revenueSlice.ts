import { createSelector, createSlice } from '@reduxjs/toolkit'

type ChartState = {
    mode: string
}

const initialState: ChartState = {
    mode: 'Pie',
}

const revenueChartSlice = createSlice({
    name: 'revenue',
    initialState: initialState,
    reducers: {
        switchMode(state, action) {
            if (action.payload === 'Pie') {
                state.mode = 'Pie'
            }
            if (action.payload === 'Bar') {
                state.mode = 'Bar'
            }
            if (action.payload === 'Line') {
                state.mode = 'Line'
            }
        },
    },
})

export const revenueChartActions = revenueChartSlice.actions
export const revenueChartReducer = revenueChartSlice.reducer
export const revenueStateSelector = (state) => state.revenue
export const revenueChartModeSelector = createSelector(
    revenueStateSelector,
    (revenueState) => revenueState.mode
)
export const isPieRevenueChart = createSelector(
    revenueChartModeSelector,
    (mode) => mode === 'Pie'
)
export const isBarRevenueChart = createSelector(
    revenueChartModeSelector,
    (mode) => mode === 'Bar'
)
export const isLineRevenueChart = createSelector(
    revenueChartModeSelector,
    (mode) => mode === 'Line'
)
