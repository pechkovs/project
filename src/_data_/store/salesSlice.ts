import { createSelector, createSlice } from '@reduxjs/toolkit'

type ChartState = {
    mode: string
}

const initialState: ChartState = {
    mode: 'Pie',
}

const salesChartSlice = createSlice({
    name: 'sales',
    initialState: initialState,
    reducers: {
        switchMode(state, action) {
            if (action.payload === 'Pie') {
                state.mode = 'Pie'
            }
            if (action.payload === 'Bar') {
                state.mode = 'Bar'
            }
            if (action.payload === 'Line') {
                state.mode = 'Line'
            }
        },
    },
})

export const salesChartActions = salesChartSlice.actions
export const salesChartReducer = salesChartSlice.reducer
export const salesStateSelector = (state) => state.sales
export const salesChartModeSelector = createSelector(
    salesStateSelector,
    (salesState) => salesState.mode
)
export const isPieSalesChart = createSelector(
    salesChartModeSelector,
    (mode) => mode === 'Pie'
)
export const isBarSalesChart = createSelector(
    salesChartModeSelector,
    (mode) => mode === 'Bar'
)
export const isLineSalesChart = createSelector(
    salesChartModeSelector,
    (mode) => mode === 'Line'
)
