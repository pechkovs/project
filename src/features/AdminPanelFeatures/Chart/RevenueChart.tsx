import React, { useState } from 'react'
//import { faker } from '@faker-js/faker'
import LineChart from './LineChart'
import BarChart from './BarChart'
import PieChart from './PieChart'
import data from '../../../components/constants/data'
import { ChartWrapper } from './charts.style'
import { ChartContainer } from './charts.style'
import { StyledChart } from './charts.style'

const myChart = () => {
    const labels = data.revenueByMonths.labels.map((month) => month)
    const [revData] = useState({
        labels,
        datasets: [
            {
                label: 'Прибыль',
                data: data.revenueByMonths.data,
                borderColor: '#bf9270',
                backgroundColor: [
                    '#bf9270',
                    '#FFD166',
                    '#118AB2',
                    '#06D6A0',
                    '#EF476F',
                    '#3844ec',
                    '#667dff',
                    '#16b211',
                    '#d67c06',
                    '#d606a2',
                    '#c2ef47',
                    '#47e9ef',
                ],
            },
        ],
    })

    const [ActiveChart, setActiveChart] = useState('Pie')

    return (
        <>
            <StyledChart>
                <button onClick={() => setActiveChart('Bar')}>Bar</button>
                <button onClick={() => setActiveChart('Line')}>Line</button>
                <button onClick={() => setActiveChart('Pie')}>Pie</button>
                <ChartContainer>
                    {ActiveChart === 'Pie' && (
                        <ChartWrapper>
                            <PieChart
                                chartData={revData}
                                chartTitle={`${'Прибыль'}`}
                            />
                        </ChartWrapper>
                    )}
                    {ActiveChart === 'Bar' && (
                        <ChartWrapper>
                            <BarChart
                                chartData={revData}
                                chartTitle={`${'Прибыль'}`}
                            />
                        </ChartWrapper>
                    )}

                    {ActiveChart === 'Line' && (
                        <ChartWrapper>
                            <LineChart
                                chartData={revData}
                                chartTitle={`${'Прибыль'}`}
                            />
                        </ChartWrapper>
                    )}
                </ChartContainer>
            </StyledChart>
        </>
    )
}

export default myChart
