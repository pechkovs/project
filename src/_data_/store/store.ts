import { combineReducers, configureStore } from '@reduxjs/toolkit'
import { rtkForm } from '../rtkform/rtkform'
import { rtkCake } from '../rtkcake/rtkcake'
import { api } from './api'
import { ordersChartReducer } from './ordersSlice'
import { revenueChartReducer } from './revenueSlice'
import { salesChartReducer } from './salesSlice'

const rootReducer = combineReducers({
    //userReducer,
    [rtkForm.reducerPath]: rtkForm.reducer,
    [api.reducerPath]: api.reducer,
    [rtkCake.reducerPath]: rtkCake.reducer,
    sales: salesChartReducer,
    revenue: revenueChartReducer,
    orders: ordersChartReducer,
})

export const setupStore = () => {
    return configureStore({
        reducer: rootReducer,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware().concat(
                rtkForm.middleware,
                api.middleware,
                rtkCake.middleware
            ),
    })
}

export const store = setupStore()

export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']
