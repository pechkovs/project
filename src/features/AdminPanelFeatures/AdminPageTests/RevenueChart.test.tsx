import { describe, it, expect } from '@jest/globals'
import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import RevenueChart from '../Chart/RevenueChart'
import { store } from '../../../_data_/store/store'
import Provider from 'react-redux/es/components/Provider'

describe('OrdersChart', () => {
    it('should render the SalesChart', () => {
        const { container } = render(
            <Provider store={store}>
                <RevenueChart />
            </Provider>
        )
        expect(container).toMatchSnapshot()
    })
})
