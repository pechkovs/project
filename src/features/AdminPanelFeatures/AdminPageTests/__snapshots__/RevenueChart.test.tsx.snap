// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersChart should render the SalesChart 1`] = `
<div>
  <div>
    loading
  </div>
</div>
`;
